core/src/test/java/io/grpc/internal/SpiffeUtilTest.java

```diff
@@ -193,4 +210,184 @@ public void spiffePathFormatTest() {
+    private static final String SPIFFE_TRUST_BUNDLE_WRONG_MULTI_CERTS =
+        "spiffebundle_wrong_multi_certs.json";
+    private static final String SPIFFE_TRUST_BUNDLE_DUPLICATES = "spiffebundle_duplicates.json";
+    private static final String SPIFFE_TRUST_BUNDLE_WRONG_ROOT = "spiffebundle_wrong_root.json";
+    private static final String SPIFFE_TRUST_BUNDLE_WRONG_SEQ = "spiffebundle_wrong_seq_type.json";
+    private static final String DOMAIN_ERROR_MESSAGE =
+        " Certificate loading for trust domain 'google.com' failed.";
+
+
+    private final TemporaryFolder tempFolder = new TemporaryFolder();
+    private String tempSpiffeTrustBundle;
+    private String tempSpiffeTrustBundleMalformed;
+    private String tempSpiffeTrustBundleCorruptedCert;
+    private String tempSpiffeTrustBundleWrongKty;
+    private String tempSpiffeTrustBundleWrongKid;
+    private String tempSpiffeTrustBundleWrongUse;
+    private String tempSpiffeTrustBundleWrongMultiCerts;
+    private String tempSpiffeTrustBundleDuplicates;
+    private String tempSpiffeTrustBundleWrongRoot;
+    private String tempSpiffeTrustBundleWrongSeq;
+
+
+    private X509Certificate[] spiffeCert;
+    private X509Certificate[] multipleUriSanCert;
+    private X509Certificate[] serverCert0;
+
+    @Before
+    public void setUp() throws Exception {
+      tempFolder.create();
+      spiffeCert = CertificateUtils.getX509Certificates(TlsTesting.loadCert(SPIFFE_PEM_FILE));
+      multipleUriSanCert = CertificateUtils.getX509Certificates(TlsTesting
+          .loadCert(MULTI_URI_SAN_PEM_FILE));
+      serverCert0 = CertificateUtils.getX509Certificates(TlsTesting.loadCert(SERVER_0_PEM_FILE));
+
+      tempSpiffeTrustBundle = copyFileToTmp(SPIFFE_TRUST_BUNDLE);
+      tempSpiffeTrustBundleMalformed = copyFileToTmp(SPIFFE_TRUST_BUNDLE_MALFORMED);
+      tempSpiffeTrustBundleCorruptedCert = copyFileToTmp(SPIFFE_TRUST_BUNDLE_CORRUPTED_CERT);
+      tempSpiffeTrustBundleWrongKty = copyFileToTmp(SPIFFE_TRUST_BUNDLE_WRONG_KTY);
+      tempSpiffeTrustBundleWrongKid = copyFileToTmp(SPIFFE_TRUST_BUNDLE_WRONG_KID);
+      tempSpiffeTrustBundleWrongUse = copyFileToTmp(SPIFFE_TRUST_BUNDLE_WRONG_USE);
+      tempSpiffeTrustBundleWrongMultiCerts = copyFileToTmp(SPIFFE_TRUST_BUNDLE_WRONG_MULTI_CERTS);
+      tempSpiffeTrustBundleDuplicates = copyFileToTmp(SPIFFE_TRUST_BUNDLE_DUPLICATES);
+      tempSpiffeTrustBundleWrongRoot = copyFileToTmp(SPIFFE_TRUST_BUNDLE_WRONG_ROOT);
+      tempSpiffeTrustBundleWrongSeq = copyFileToTmp(SPIFFE_TRUST_BUNDLE_WRONG_SEQ);
+    }
+
+    private String copyFileToTmp(String fileName) throws Exception {
+      InputStream resourceStream = SpiffeUtilTest.class.getClassLoader()
+          .getResourceAsStream(TEST_DIRECTORY_PREFIX + fileName);
+      Path tempFilePath = tempFolder.newFile(fileName).toPath();
+      Files.copy(resourceStream, tempFilePath, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
This doesn't close the InputStream. Create `resourceStream` with try-with-resources.